build/elasticsearch/bin/es-docker

```diff
@@ -22,4 +22,6 @@ do
     fi
 done < <(env)
 
+echo "-Des.cgroups.hierarchy.override=/" >> config/jvm.options
```

Comment:
I like this much more. Mutating that file could be really nasty, since it might be owned by the user and living on some persistent storage that they control.

If we can apply this fix via an environment variable, is there any reason not to do so with an `ENV` declaration in the Dockerfile? I think that's the most transparent and obvious place for future readers to understand what's happening.

Wherever we do it, let's also have a comment linking back to this thread, since it's a pretty subtle thing that's going on.